### PR TITLE
Vcref projection

### PR DESCRIPTION
**Pull request by marclaval**
Only the second commit is to be reviewed, the first one comes from #23189

### CODE REVIEW

**Review by mhevery — APPROVED** [review]

**Comment by vicb** [comment]
merge-assistance -> depends on 23189

File: packages/core/src/render3/node_manipulation.ts [hunk 1]
```diff
@@ -490,4 +490,7 @@ export function appendProjectedNode(
       addRemoveViewFromContainer(node as LContainerNode, views[i], true, null);
     }
   }
+  if (node.dynamicLContainerNode) {
+    node.dynamicLContainerNode.data.renderParent = currentParent as LElementNode;
```
Comment (vicb) [inline packages/core/src/render3/node_manipulation.ts]: By looking at this is code, it is not clear to me why `currentParent` is a `LElementNode`.

Add a comment and/or assert ?

File: packages/core/test/render3/view_container_ref_spec.ts [hunk 2]
```diff
@@ -6,39 +6,39 @@
...
+      type: DirectiveWithVCRef,
+      selectors: [['', 'vcref', '']],
+      factory: () => directiveInstance = new DirectiveWithVCRef(injectViewContainerRef()),
+      inputs: {tplRef: 'tplRef'}
+    });
+
+    tplRef: TemplateRef<{}>;
 
-    beforeEach(() => { directiveInstance = null; });
+    constructor(public vcref: ViewContainerRef) {}
```
Comment (vicb) [inline packages/core/test/render3/view_container_ref_spec.ts]: suggestion: Add a comment in the lines of "injecting a ViewContainerRef to create a dynamic blablabla"

File: packages/core/test/render3/view_container_ref_spec.ts [hunk 3]
```diff
@@ -452,4 +452,188 @@ describe('ViewContainerRef', () => {
...
+      const fixture = new ComponentFixture(Parent);
+      expect(fixture.html).toEqual('<child><div><header vcref="">blah</header></div></child>');
+
+      directiveInstance !.vcref.createEmbeddedView(directiveInstance !.tplRef, fixture.component);
+      fixture.update();
+      expect(fixture.html)
+          .toEqual('<child><div><header vcref="">blah</header><span>bar</span></div></child>');
+    });
+
+    @Component({
```
Comment (vicb) [inline packages/core/test/render3/view_container_ref_spec.ts]: ThisSeems to belong in the `it()` below

**Review by vicb — CHANGES_REQUESTED** [review]
see inline comments

**Comment by IgorMinar** [comment]
I rebased this and force-pushed

**Comment by marclaval** [comment]
@vicb the changes that you requested are in #23287

**Pull request closed without merging**